src/datasets/packaged_modules/csv/csv.py

```diff
@@ -145,7 +145,7 @@ def _split_generators(self, dl_manager):
         """We handle string, list and dicts in datafiles"""
         if not self.config.data_files:
             raise ValueError(f"At least one data file must be specified, but got data_files={self.config.data_files}")
-        data_files = dl_manager.download_and_extract(self.config.data_files)
+        data_files = dl_manager.download_and_extract(self.config.data_files, glob_archives=True)
```

Comment:
there's already some globbing a few lines after with
```python
            if any(os.path.isdir(file) for file in files):
                files = [file for file in _iter_files(files)]
```
Could it be possible to make `_iter_files` work with streaming, instead of having an additional logic ?

Reply:
Let me think of a solution...

Reply:
Please also note that you implemented the functionality in the function `_iter_files` (not a method) and use it only in "csv" module, whereas that functionality is needed in all packaged modules ("json", "text",...).

My approach was to put that functionality inside the DownloadManager, so that it can be automatically used in all packaged modules ("csv", "json",...).

Therefore, the ideal solution is:
- to merge the functionality of `_iter_files` and `glob_archive`: so that we do not have redundant calls to glob
- and put it inside the DownloadManager: so that this functionality can be used in all packages modules ("csv", "json"...)

Reply:
That makes sense

If it can help there's also a PR that adds support for ZIP archives for `iter_archive`, if it can be helpful: #3379. The `iter_archive` method should also be better in terms of performance for streaming thanks to buffering, compared to globbing + opening the inner files one by one. If you think this can be useful I can help the contributor to merge this PR faster :)

Reply:
Proposed approach: implement `[Streaming]DownloadManager.iter_files` using `os.walk` [`xwalk`].